### PR TITLE
Switch to github actions

### PR DESCRIPTION
**Pull request by ghiculescu**
This repo is experiencing pretty large CI backlogs at the moment. I'm guessing they are caused by issues @ Travis CI: https://news.ycombinator.com/item?id=25338983

This PR switches the CI over to Github actions, and removes Travis CI.

You can see outputs at https://github.com/ghiculescu/cancancan/actions (or at least I can). I think you'll need to [enable something](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/disabling-or-limiting-github-actions-for-a-repository) for the checks to show against this PR.

Other changes:

- I dropped Ruby 2.3 support. It's been EOL'd for over a year. https://www.ruby-lang.org/en/downloads/branches/
- I included a comment about this, but it's not clear to me why we have a failing test on jruby / AR 5.2. I haven't investigated yet. On Travis this combination was also allowed to fail, so I've excluded it in the test matrix here.
- I've included the changes in https://github.com/CanCanCommunity/cancancan/pull/657 to run tests against Rails master. I've also added Rails 6.1. To get a green build here, I've included the single code change from #657.
- I just [can't](https://github.com/ghiculescu/cancancan/runs/1533587698?check_suite_focus=true) get Nokogiri to work on Truffleruby, so I've excluded it until someone can figure that out.
- Same goes for Jruby + Rails 6.1. [Here's](https://github.com/ghiculescu/cancancan/runs/1533659113?check_suite_focus=true) a sample failure. [Rails 6 is fine](https://github.com/ghiculescu/cancancan/runs/1533659099?check_suite_focus=true). 

image

### CODE REVIEW

**Comment by ghiculescu** [comment]
@coorasse this is ready to go, what do you think?

**Comment by ghiculescu** [comment]
image
@coorasse FYI

**Comment by coorasse** [comment]
I am very happy of switching to GitHub Actions. I have a personal preference for SemaphoreCI but they lack support for build matrixes and `exclude` some combinations. 😢 
Regarding Truffleruby I'd like to keep it running. @eregon can you help @ghiculescu with the nokogiri issue?

**Comment by eregon** [comment]
See https://github.com/oracle/truffleruby/issues/62, so just adding this should fix it:
```yaml
    - run: sudo apt-get -yqq install libxml2-dev libxslt-dev
      if: startsWith(matrix.ruby, 'truffleruby')
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          ruby-version: ${{ matrix.ruby }}
+
+      - name: Install bundler
+        run: gem install bundler
+
+      # https://github.com/oracle/truffleruby/issues/62#issuecomment-362065726
+      # TODO: can't get this to work on Github actions. see comments in https://github.com/CanCanCommunity/cancancan/pull/669
+      - name: Nokogiri support for Truffleruby
+        run: BUNDLE_GEMFILE=${{ matrix.gemfile }} bundle config build.nokogiri --use-system-libraries
+        if: ${{ matrix.ruby == 'truffleruby' }}
```
Comment (eregon) [inline .github/workflows/test.yml:67]: This is unnecessary, TruffleRuby automatically uses system libraries for nokogiri.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+      - name: Install gems
+        run: BUNDLE_GEMFILE=${{ matrix.gemfile }} bundle install --jobs 2
```
Comment (eregon) [inline .github/workflows/test.yml:70]: It's better to set `BUNDLE_GEMFILE` in the `env:` of the job, as described in https://github.com/ruby/setup-ruby#caching-bundle-install-automatically

**Comment by ghiculescu** [comment]
Thanks for your help @eregon!

@coorasse we are now green on Truffleruby: https://github.com/ghiculescu/cancancan/actions/runs/431036514

**Comment by coorasse** [comment]
@ghiculescu could you please re-enable also Travis in this PR? I'd like to run both of them until I have a green build on Github Actions. Afterwards I'll remove the Travis configuration. Thank you! 🙏

**Comment by coorasse** [comment]
or maybe is enough to rebase and commit. I enabled Actions now.

**Comment by ghiculescu** [comment]
Hmmm. That doesn't seem to have worked :/

**Comment by ghiculescu** [comment]
> @ghiculescu could you please re-enable also Travis in this PR? 

Done @coorasse

**Review by coorasse — APPROVED** [review]

**Comment by coorasse** [comment]
Nice! The first build has started 😄 https://github.com/CanCanCommunity/cancancan/actions/runs/452447924

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Install bundler
+        run: gem install bundler
+
+      # https://github.com/CanCanCommunity/cancancan/pull/669#issuecomment-748019539
+      - name: Nokogiri support for Truffleruby
+        run: sudo apt-get -yqq install libxml2-dev libxslt-dev
+        if: ${{ matrix.ruby == 'truffleruby' }}
+
+      - name: Install gems
+        run: bundle install --jobs 2
```
Comment (eregon) [inline .github/workflows/test.yml:72]: Using `bundler-cache: true` would be much faster, I'll make a PR for that.

Comment (eregon) [reply .github/workflows/test.yml:72]: https://github.com/CanCanCommunity/cancancan/pull/682

**Pull request merged**